modules/light-clients/07-tendermint/upgrade.go

```diff
@@ -165,3 +174,19 @@ func constructUpgradeConsStateMerklePath(upgradePath []string, lastHeight export
 
 	return commitmenttypes.NewMerklePath(consStateKey...)
 }
+
+// calculateNewTrustingPeriod converts the provided durations to decimal representation to avoid floating-point precision issues
+// and calculates the new trusting period, decreasing it by the ratio between the original and new unbonding period.
+func calculateNewTrustingPeriod(trustingPeriod, originalUnbonding, newUnbonding time.Duration) time.Duration {
+	origUnbondingDec := sdkmath.LegacyNewDec(originalUnbonding.Nanoseconds())
+	newUnbondingDec := sdkmath.LegacyNewDec(newUnbonding.Nanoseconds())
+	trustingPeriodDec := sdkmath.LegacyNewDec(trustingPeriod.Nanoseconds())
+
+	// compute decrease ratio: (originalUnbonding - newUnbonding) / originalUnbonding
+	decreaseRatio := origUnbondingDec.Sub(newUnbondingDec).Quo(origUnbondingDec)
+
+	// compute new trusting period: trustingPeriod * (1 - decreaseRatio)
+	newTrustingPeriodDec := trustingPeriodDec.Mul(sdkmath.LegacyOneDec().Sub(decreaseRatio))
```

Comment:
Isn't it much simpler to just do:


```
newTrustingPeriodDec := trustingPeeriodDec.Mul(newUnbondingDec).Quo(origUnbondingDec)
```

This is equivalent to what you have if I'm not mistaken

Reply:
lol, yes, I think they are equivalent. Happy to use this if you find it more readable. I can update!

Reply:
updated in https://github.com/cosmos/ibc-go/pull/8185/commits/bfedb3d57c53a929f86d9c66b760d8d2dd71efed